src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -66,8 +90,7 @@ public GetShardIteratorResult getShardIterator(GetShardIteratorRequest request)
      * stream shard, so that clients can request a new shard iterator where they left off.
      */
     @Override
-    protected MtGetRecordsResult getRecords(GetRecordsRequest request, Function<Record, MtRecord> recordMapper,
-                                            Predicate<MtRecord> recordFilter) {
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, MtStreamArn mtStreamArn) {
```

Comment:
If we are called with tenant context, we need to filter records to those that have matching context, but then we can use a cached record mapper to convert them.